src/jltypes.c

```diff
@@ -2416,12 +2515,15 @@ static jl_value_t *inst_type_w_(jl_value_t *t, jl_typeenv_t *env, jl_typestack_t
         jl_value_t *N = NULL;
         JL_GC_PUSH2(&T, &N);
         if (v->T) {
-            T = inst_type_w_(v->T, env, stack, check);
-            if (v->N)
-                N = inst_type_w_(v->N, env, stack, check);
+            T = inst_type_w_(v->T, env, stack, check, nothrow);
+            if (T == NULL)
+                T = jl_bottom_type;
+            if (v->N) // This branch should never throw.
+                N = inst_type_w_(v->N, env, stack, check, 0);
```

Comment:
FWIW, this branch does potentially throw
```
julia> Tuple{Vararg{Int,Union{N,Complex{N}}}} where N
NTuple{Union{Complex{N}, N}, Int64} where N

julia> ans{String}
ERROR: TypeError: in Complex, in T, expected T<:Real, got Type{String}
Stacktrace:
 [1] top-level scope
   @ REPL[6]:1
```

Reply:
My bad, shouldn't assume invalid type don't exist. (IIRC, only Int and TypeVar with no constraint could be used as vararg length)